ZenPacks/zenoss/LinuxMonitor/parsers/linux/diskstats.py

```diff
@@ -1,140 +1,190 @@
 
-        # save global zHardDiskMapMatch filtering regex property, 
-        # to be copied to each datapoint's .data attribute
-        ret['hdFilterRegex'] = getattr(context, self.HARD_DISK_MATCH_PROPERTY, None)
+LOG = logging.getLogger('zen.command.parsers.diskstats')
 
+
+class diskstats(ComponentCommandParser):
+
+    scanner = r''.join([COMPONENT, TOTAL_READS, NUM_READS_MERGED,
+                        TOTAL_SECTORS_READ, TOTAL_MS_READING,
+                        TOTAL_WRITES, NUM_WRITES_MERGED,
+                        TOTAL_SECTORS_WRITTEN, TOTAL_MS_WRITING,
+                        NUM_IO_IN_PROGESS, TOTAL_MS_IO, WEIGHTED_MS_PER_IO])
+    splitter = '\n'
+
+    def dataForParser(self, context, datapoint):
+        ret = super(diskstats, self).dataForParser(context, datapoint)
+        ret['hdFilterRegex'] = getattr(context, 'zHardDiskMapMatch', COMPONENT)
         return ret
 
     def processResults(self, cmd, result):
-        """
-        Given "/proc/diskstats" output, filter out any logical devices and sum
-        the total read sectors and write sectors for all physical devices. 
-        Assign the total read sectors to ssIORawReceived and the total write
-        sectors to ssIORawSent. Here is an example of the command's output:
-
-        1    0 ram0 0 0 0 0 0 0 0 0 0 0 0
-        8    0 sda 10541549 14506876 290280380 125470349 29042916 90285293 954627734 801503911 0 134474409 926975991
-        
-        in this case [290280380, 954627734] would be returned.
-        """
-        pointsMap = dict((dp.id, dp) for dp in cmd.points)
-        destkeys = [key for key in OUTPUT_DATA_KEYS if key in pointsMap]
-
-        # only bother parsing if there is some data of interest to be parsed
-        if destkeys:
-
-            # see if a filtering regex was attached to the datapoints - just look
-            # at the first point, all will have the same value attached
-            hdRegex= cmd.points[0].data['hdFilterRegex']
-            log.debug("using hdRegex: %s", hdRegex)
-            hdFilterFn = None
-            if hdRegex:
-                try:
-                    hdFilterFn = _getReMatcher(hdRegex)
-                except Exception as e:
-                    log.warning("%s ignored, invalid regex specified: %s", self.HARD_DISK_MATCH_PROPERTY, e)
-
-            # parse command output to extract values
-            log.debug("about to parse: %s", cmd.result.output)
-            valueMap = _parseOutput(cmd.result.output, hdFilterFn)
-
-            # save (datapoint, value) tuples to result values
-            result.values.extend((pointsMap[key], valueMap[key]) for key in destkeys)
-            log.debug("parsed values: %s", zip(destkeys, (x[1] for x in result.values[-len(destkeys):])))
+        datapointMap = dict((dp.id, dp) for dp in cmd.points)
+        if any((x in datapointMap for x in PARTITION_DATAPOINTS)):
+            hdRegex = re.compile(cmd.points[0].data['hdFilterRegex'])
```

Comment:
Not safe to `re.compile()` a user-defined regex without catching the exception. Note how this is done in the current version of diskstats.py.